pyvista/core/utilities/cell_quality.py

```diff
@@ -0,0 +1,325 @@
+INF = float('inf')
+ANGLE = float((180 / np.pi) * np.arccos(1 / 3))
+R22 = sqrt(2) / 2
+R33 = sqrt(3) / 3
+
+TRIANGLE: Literal[CellType.TRIANGLE] = CellType.TRIANGLE
+QUAD: Literal[CellType.QUAD] = CellType.QUAD
+TETRA: Literal[CellType.TETRA] = CellType.TETRA
+HEXAHEDRON: Literal[CellType.HEXAHEDRON] = CellType.HEXAHEDRON
+PYRAMID: Literal[CellType.PYRAMID] = CellType.PYRAMID
+WEDGE: Literal[CellType.WEDGE] = CellType.WEDGE
+
+Info = CellQualityInfo
+
+_CELL_QUALITY_INFO = [
+    Info(TRIANGLE, 'area', (0.0, INF), (0.0, INF), (0.0, INF), sqrt(3.0) / 4.0),
+    Info(TRIANGLE, 'aspect_ratio', (1.0, 1.3), (1.0, INF), (1.0, INF), 1.0),
+    Info(TRIANGLE, 'aspect_frobenius', (1.0, 1.3), (1.0, INF), (1.0, INF), 1.0),
+    Info(TRIANGLE, 'condition', (1.0, 1.3), (1.0, INF), (1.0, INF), 1.0),
+    Info(TRIANGLE, 'distortion', (0.5, 1.0), (0.0, 1.0), (-INF, INF), 1.0),
+    Info(TRIANGLE, 'max_angle', (60.0, 90.0), (60.0, 180.0), (0.0, 180.0), 60.0),
+    Info(TRIANGLE, 'min_angle', (30.0, 60.0), (0.0, 60.0), (0.0, 360.0), 60.0),
+    Info(TRIANGLE, 'scaled_jacobian', (0.5, 2 * R33), (-2 * R33, 2 * R33), (-INF, INF), 1.0),
+    Info(TRIANGLE, 'radius_ratio', (1.0, 3.0), (1.0, INF), (1.0, INF), 1.0),
+    Info(TRIANGLE, 'shape', (0.25, 1.0), (0.0, 1.0), (0.0, 1.0), 1.0),
+    Info(TRIANGLE, 'shape_and_size', (0.25, 1.0), (0.0, 1.0), (0.0, 1.0), None),
+    Info(QUAD, 'area', (0.0, INF), (0.0, INF), (-INF, INF), 1.0),
+    Info(QUAD, 'aspect_ratio', (1.0, 1.3), (1.0, INF), (1.0, INF), 1.0),
+    Info(QUAD, 'condition', (1.0, 4), (1.0, INF), (1.0, INF), 1.0),
+    Info(QUAD, 'distortion', (0.5, 1.0), (0.0, 1.0), (-INF, INF), 1.0),
+    Info(QUAD, 'jacobian', (0.0, INF), (0.0, INF), (-INF, INF), 1.0),
+    Info(QUAD, 'max_aspect_frobenius', (1.0, 1.3), (1.0, INF), (1.0, INF), 1.0),
+    Info(QUAD, 'max_angle', (90.0, 135.0), (90.0, 360.0), (0.0, 360.0), 90.0),
+    Info(QUAD, 'max_edge_ratio', (1.0, 1.3), (1.0, INF), (1.0, INF), 1.0),
+    Info(QUAD, 'med_aspect_frobenius', (1.0, 1.3), (1.0, INF), (1.0, INF), 1.0),
+    Info(QUAD, 'min_angle', (45.0, 90.0), (0.0, 90.0), (0.0, 360.0), 90.0),
+    Info(QUAD, 'oddy', (0.0, 0.5), (0.0, INF), (0.0, INF), 0.0),
+    Info(QUAD, 'radius_ratio', (1.0, 1.3), (1.0, INF), (1.0, INF), 1.0),
+    Info(QUAD, 'relative_size_squared', (0.3, 1.0), (0.0, 1.0), (0.0, 1.0), None),
+    Info(QUAD, 'scaled_jacobian', (0.3, 1.0), (-1.0, 1.0), (-1.0, 1.0), 1.0),
+    Info(QUAD, 'shape', (0.3, 1.0), (0.0, 1.0), (0.0, 1.0), 1.0),
+    Info(QUAD, 'shape_and_size', (0.2, 1.0), (0.0, 1.0), (0.0, 1.0), None),
+    Info(QUAD, 'shear', (0.3, 1.0), (0.0, 1.0), (0.0, 1.0), 1.0),
+    Info(QUAD, 'shear_and_size', (0.2, 1.0), (0.0, 1.0), (0.0, 1.0), None),
+    # Info(QUAD, 'skew', (0.5, 1.0), (0.0, 1.0), (0.0, 1.0), 1.0),
+    Info(QUAD, 'stretch', (0.25, 1.0), (0.0, 1.0), (0.0, INF), 1.0),
+    Info(QUAD, 'taper', (0.0, 0.7), (0.0, INF), (0.0, INF), 0.0),
+    # Info(QUAD, 'warpage', (0.0, 0.7), (0.0, 2.0), (0.0, INF), 0.0),
```

Comment:
`'skew'` and `'warpage'` seem to be buggy or at least contradict the values reported by the reference manual since:
- `'skew'` returns `0.0`, not `1.0` for a unit cell
-  `'warpage'` returns `1.0`, not `0.0` for a unit cell

``` python
import pyvista as pv
quad = pv.examples.cells.Quadrilateral()

skew = quad.cell_quality('skew').active_scalars[0]
print(skew) # 0.0

warpage = quad.cell_quality('warpage').active_scalars[0]
print(warpage) # 1.0
```

Maybe the these should be flipped ? i.e. 
```suggestion
    Info(QUAD, 'warpage', (0.5, 1.0), (0.0, 1.0), (0.0, 1.0), 1.0),
    Info(QUAD, 'stretch', (0.25, 1.0), (0.0, 1.0), (0.0, INF), 1.0),
    Info(QUAD, 'taper', (0.0, 0.7), (0.0, INF), (0.0, INF), 0.0),
    Info(QUAD, 'skew', (0.0, 0.7), (0.0, 2.0), (0.0, INF), 0.0),
```